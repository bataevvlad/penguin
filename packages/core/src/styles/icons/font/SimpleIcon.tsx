import {useCallback, useMemo} from 'react';
import {TextStyle} from 'react-native';

import Text from '../../../basic/components/Text/Text';
import useTheme from '../../hooks/useTheme';
import px from '../../utils/px';

import glyphMap from './unicodesMap.json';

interface IProps {
  name: SimpleIconName;
  color?: string | undefined;
  size?: number;
  style?: TextStyle;
}

const SimpleIcon = ({
  size = px(24),
  name,
  style,
  color,
}: IProps): JSX.Element => {
  const {colors} = useTheme();
  const resolveGlyph = useCallback((iconName: SimpleIconName): string => {
    const glyph = glyphMap[iconName];
    return String.fromCodePoint(glyph);
  }, []);

  const icon = useMemo(() => resolveGlyph(name), [name]);

  const textStyle = {
    fontSize: size,
    fontFamily: 'Neotis',
    color: color ?? colors.IconNeutral,
    ...style,
  };
  return <Text style={textStyle}>{icon}</Text>;
};

export type SimpleIconName = keyof typeof glyphMap;

export const iconNames = Object.keys(glyphMap) as SimpleIconName[];

export default SimpleIcon;
