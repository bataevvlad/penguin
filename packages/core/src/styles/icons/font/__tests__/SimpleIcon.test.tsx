import {render} from '@testing-library/react-native';

import SimpleIcon from '../SimpleIcon';

describe('@penguin/core/icons/SimpleIcon', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<SimpleIcon name={'icon-account'} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
