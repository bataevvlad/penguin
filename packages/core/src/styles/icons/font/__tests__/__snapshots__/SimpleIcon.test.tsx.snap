// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@penguin/core/icons/SimpleIcon renders correctly 1`] = `
<Text
  allowFontScaling={true}
  style={
    {
      "color": "#353A45",
      "fontFamily": "Neotis",
      "fontSize": 24,
    }
  }
>
  î¨Ž
</Text>
`;
